plugins/flatpak/gs-flatpak.c

```diff
@@ -323,6 +323,22 @@ gs_flatpak_add_flatpak_keyword_cb (XbBuilderFixup *self,
 	return TRUE;
 }
 
+static gboolean
+gs_flatpak_mark_endlessm_apps_popular_cb (XbBuilderFixup *self,
+					   XbBuilderNode *bn,
+					   gpointer user_data,
+					   GError **error)
+{
+	if (g_strcmp0 (xb_builder_node_get_element (bn), "component") == 0) {
+		g_autoptr(XbBuilderNode) id = xb_builder_node_get_child (bn, "id", NULL);
+		if (g_str_has_prefix (xb_builder_node_get_text (id), "com.endlessm.") ||
+		    g_str_has_prefix (xb_builder_node_get_text (id), "com.endlessnetwork.")) {
```

Comment:
It’s possible for `xb_builder_node_get_text()` to return `NULL` (if the XML contains `<id/>` or `<id></id>`), which `g_str_has_prefix()` won’t be happy with.